src/upload.test.js

```diff
@@ -1,97 +1,155 @@
+
+    expect(mock.history.post.length).toBe(1);
+    const request = mock.history.post[0];
+    mock.resetHistory();
+
+    expect(request.auth).toEqual({ username: "", password: "foobar" });
+    await compareFormData(request.data, [["file", "foo", filename]]);
+
+    expect(data).toEqual({ skylink });
+  });
+
+  it("should send custom user agent if defined", async () => {
+    const client2 = new SkynetClient(portalUrl, { customUserAgent: "Sia-Agent" });
+
+    // Should use client's user agent.
+
+    let data = await client2.upload(file);
+
+    expect(mock.history.post.length).toBe(1);
+    let request = mock.history.post[0];
+    mock.resetHistory();
+
+    expect(request.headers["User-Agent"]).toEqual("Sia-Agent");
+    // Check that other headers weren't altered.
+    expect(request.headers["Content-Type"]).toEqual("application/x-www-form-urlencoded");
+    await compareFormData(request.data, [["file", "foo", filename]]);
+
+    expect(data).toEqual({ skylink });
+
+    // Should use user agent set in options to function.
+
+    data = await client2.upload(file, { customUserAgent: "Sia-Agent-2" });
+
+    expect(mock.history.post.length).toBe(1);
+    request = mock.history.post[0];
+    mock.resetHistory();
+
+    expect(request.headers["User-Agent"]).toEqual("Sia-Agent-2");
+    // Check that other headers weren't altered.
+    expect(request.headers["Content-Type"]).toEqual("application/x-www-form-urlencoded");
+    await compareFormData(request.data, [["file", "foo", filename]]);
 
     expect(data).toEqual({ skylink });
   });
 });
 
 describe("uploadDirectory", () => {
-  const blob = new Blob([], { type: "image/jpeg" });
   const filename = "i-am-root";
   const directory = {
-    "i-am-not/file1.jpeg": new File([blob], "i-am-not/file1.jpeg"),
-    "i-am-not/file2.jpeg": new File([blob], "i-am-not/file2.jpeg"),
-    "i-am-not/me-neither/file3.jpeg": new File([blob], "i-am-not/me-neither/file3.jpeg"),
+    "i-am-not/file1.jpeg": new File(["foo1"], "i-am-not/file1.jpeg"),
+    "i-am-not/file2.jpeg": new File(["foo2"], "i-am-not/file2.jpeg"),
+    "i-am-not/me-neither/file3.jpeg": new File(["foo3"], "i-am-not/me-neither/file3.jpeg"),
   };
+  const url = `${portalUrl}/skynet/skyfile?filename=${filename}`;
 
   beforeEach(() => {
-    axios.post.mockResolvedValue({ data: { skylink } });
+    mock.onPost(url).reply(200, { skylink: skylink });
   });
 
-  it("should send post request with FormData", () => {
-    client.uploadDirectory(directory, filename);
+  it("should send formdata with files", async () => {
+    const data = await client.uploadDirectory(directory, filename);
 
-    expect(axios.post).toHaveBeenCalledWith(
-      `${portalUrl}/skynet/skyfile?filename=${filename}`,
-      expect.any(FormData), // TODO: Inspect data contents.
-      undefined
-    );
-  });
+    expect(mock.history.post.length).toBe(1);
+    const request = mock.history.post[0];
+    mock.resetHistory();
```

Comment:
reset history before the tests, not during (covered "why" in a different comment), same applies to other tests